### ISSUE TITLE
authentication doesn't work

### ISSUE THREAD

**Issue by bengdan** [django-tastypie/django-tastypie#1617]
## Error log is attached

```
{
    "error_message": "'ApiKeyAuthentication' object has no attribute 'read_list'",
    "traceback": "Traceback (most recent call last):\n\n  File \"/home/ubuntu/.local/lib/python3.8/site-packages/tastypie/resources.py\", line 227, in wrapper\n    response = callback(request, *args, **kwargs)\n\n  File \"/home/ubuntu/.local/lib/python3.8/site-packages/tastypie/resources.py\", line 467, in dispatch_list\n    return self.dispatch('list', request, **kwargs)\n\n  File \"/home/ubuntu/.local/lib/python3.8/site-packages/tastypie/resources.py\", line 499, in dispatch\n    response = method(request, **kwargs)\n\n  File \"/home/ubuntu/.local/lib/python3.8/site-packages/tastypie/resources.py\", line 1355, in get_list\n    objects = self.obj_get_list(bundle=base_bundle, **self.remove_api_resource_names(kwargs))\n\n  File \"/home/ubuntu/.local/lib/python3.8/site-packages/tastypie/resources.py\", line 2207, in obj_get_list\n    return self.authorized_read_list(objects, bundle)\n\n  File \"/home/ubuntu/.local/lib/python3.8/site-packages/tastypie/resources.py\", line 628, in authorized_read_list\n    auth_result = self._meta.authorization.read_list(object_list, bundle)\n\nAttributeError: 'ApiKeyAuthentication' object has no attribute 'read_list'\n"
}
```

>Both `BasicAuthentication` and `ApiKeyAuthentication` raise this kind of Exception.

**Comment by jayvdb** [comment]
I might be seeing the same problem with `test_apikey_and_authentication_enforce_user` and `test_is_authenticated` failing against latest version of all dependencies.

**Comment by bengdan** [comment]
> I might be seeing the same problem with `test_apikey_and_authentication_enforce_user` and `test_is_authenticated` failing against latest version of all dependencies.

do u know which version without this issue?

**Comment by jayvdb** [comment]
Django 2.2.12 doesnt have this problem

**Comment by georgedorn** [comment]
@jayvdb @bengdan Can you re-test against the current master branch?  I've just merged support for django 3.2.

**Comment by georgedorn** [comment]
Closing; tests are passing against lastest tastypie; be sure to use the right version to go with your version of django.
